GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -126,34 +126,26 @@ public static bool TryGetVersion(string gitBinPath, out GitVersion gitVersion, o
             return true;
         }
 
-        public bool TryKillRunningProcess()
+        public bool TryKillRunningProcess(out string processName, out int exitCode, out string error)
         {
             this.stopping = true;
+            processName = string.Empty;

```

Comment:
I think other places in the project generally return `null` when error is not set. What does a default exit code of `-1` indicate?

Reply:
I was using it to indicate it hasn't been set.  The exit codes for the calling processes are >=0.  If you've got a better suggestion let me know!

Reply:
I think we use non-zero to indicate error in some places - at least it looks like at least some places in GitHubUpgrader do. I think `0` indicates no error, and so that is what I would have expected on initial read.

If we keep with `-1`, maybe add a method comment with how consumers should interpret the results from calling this method - they might only expect values >= 0?